tools/go-cli/go-whisk-cli/commands/commands.go

```diff
@@ -87,56 +87,46 @@ func init() {
     }
 }
 
+func getKeyValueArgs(args []string, argIndex int, parsedArgs []string) ([]string, []string, error) {
+    var whiskErr error
+
+    if len(args) - 1 >= argIndex + 2 {
+        parsedArgs = append(parsedArgs, args[argIndex + 1])
+        parsedArgs = append(parsedArgs, args[argIndex + 2])
+        args = append(args[:argIndex], args[argIndex + 3:]...)
+    } else {
+        whisk.Debug(whisk.DbgError, "Arguments for '%s' must be a key/value pair; args: %s", args[argIndex], args)
+        errMsg := wski18n.T("Arguments for '{{.arg}}' must be a key/value pair",
+            map[string]interface{}{"arg": args[argIndex]})
+        whiskErr = whisk.MakeWskError(errors.New(errMsg), whisk.EXITCODE_ERR_GENERAL, whisk.DISPLAY_MSG,
+            whisk.DISPLAY_USAGE)
+    }
+
+    return parsedArgs, args, whiskErr
+}
+
 func parseArgs(args []string) ([]string, []string, []string, error) {
     var paramArgs []string
     var annotArgs []string
     var whiskErr error
-    var errMsg string
-    var parsingParam bool
-    var parsingAnnot bool
 
     i := 0
 
     for i < len(args) {
-
-        if !parsingAnnot && args[i] == "-p" || args[i] == "--param" {
-            parsingParam = true
-
-            if len(args) - 1 >= i + 2 {
-                parsingParam = false
-                paramArgs = append(paramArgs, args[i + 1])
-                paramArgs = append(paramArgs, args[i + 2])
-                args = append(args[:i], args[i + 3:]...)
-            } else if len(args) - 1 >= i + 1 {
-                parsingParam = false
-                paramArgs = append(paramArgs, args[i + 1])
-                paramArgs = append(paramArgs, "")
-                args = append(args[:i], args[i + 2:]...)
-            } else {
-                whisk.Debug(whisk.DbgError, "Parameter arguments must be a key value pair; args: %s", args[i:])
-                errMsg = fmt.Sprintf(
-                    wski18n.T("Parameter arguments must be a key value pair: {{.args}}", map[string]interface{}{"args":args[i:]}))
+        if args[i] == "-p" || args[i] == "--param" {
+            paramArgs, args, whiskErr = getKeyValueArgs(args, i, paramArgs)
+            if whiskErr != nil {
+                whisk.Debug(whisk.DbgError, "getKeyValueArgs(%#v, %d) failed: %s\n", args, i, whiskErr)
+                errMsg := wski18n.T("Failed to obtain key/value: {{.err}}", map[string]interface{}{"err": whiskErr})
```

Comment:
Since a -a or -p option can be used, the message should indicate the option.  Same comment for the -a option processing below in line 129